modules/EED_SquareOnsiteOAuth.module.php

```diff
@@ -721,37 +734,81 @@ public static function decryptString(string $text, bool $sandbox_mode): string
     }
 
 
+    /**
+     * Clean the array of data from sensitive information.
+     *
+     * @param array $data
+     * @return array
+     */
+    private static function cleanDataArray(array $data): array
+    {
+        $sensitive_data = [
+            'access_token',
+            'refresh_token',
+            'nonce',
+        ];
+        foreach ($data as $name => $value) {
+            // for multi. arrays
+            if (is_array($value)) {
+                foreach ($value as $key => $val) {
+                    if (in_array($key, $sensitive_data)) {
+                        unset($data[ $key ]);
+                    }
+                }
```

Comment:
mmmm... i think you might have meant:

```suggestion
                foreach ($value as $key => $val) {
                    if (in_array($key, $sensitive_data)) {
                        unset($value[ $key ]);
                    }
                }
```

`$key` will surely not exist in the `$data` array since `$key` is coming from the inner array and `$data` is the parent array

Reply:
Oh, thanks for the `array_diff_key()` ! Didn't think of that one 🤦‍♂️

And yes, I was planning to use recursion and somehow dind't.